src/DynamicData/Cache/ObservableCacheEx.cs

```diff
@@ -5049,6 +5049,86 @@ public static IObservable<IReadOnlyCollection<T>> StartWithEmpty<T>(this IObserv
         return new TreeBuilder<TObject, TKey>(source, pivotOn, predicateChanged).Run();
     }
 
+    /// <summary>
+    /// Projects each update item to a new form using the specified transform function and when an update is received, allows the preservation of the previous instance.
+    /// </summary>
+    /// <typeparam name="TDestination">The type of the destination.</typeparam>
+    /// <typeparam name="TSource">The type of the source.</typeparam>
+    /// <typeparam name="TKey">The type of the key.</typeparam>
+    /// <param name="source">The source.</param>
+    /// <param name="transformFactory">The transform factory.</param>
+    /// <param name="updateAction">The update action.</param>
+    /// <returns>
+    /// A transformed update collection.
+    /// </returns>
+    /// <exception cref="System.ArgumentNullException">source
+    /// or
+    /// transformFactory.</exception>
+    public static IObservable<IChangeSet<TDestination, TKey>> TransformWithInlineUpdate<TDestination, TSource, TKey>(this IObservable<IChangeSet<TSource, TKey>> source, Func<TSource, Optional<TSource>, TKey, TDestination> transformFactory, Action<TDestination, TSource> updateAction)
```

Comment:
You could combine these 2 overloads into one by making the error handler an optional param.